compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/dnd/ExternalDrag.desktop.kt

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.dnd
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.toPainter
+import androidx.compose.ui.layout.LayoutCoordinates
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.window.LocalWindow
+import androidx.compose.ui.window.density
+import java.awt.Image
+import java.awt.Point
+import java.awt.Window
+import java.awt.datatransfer.DataFlavor
+import java.awt.datatransfer.DataFlavor.selectBestTextFlavor
+import java.awt.datatransfer.Transferable
+import java.awt.dnd.DnDConstants
+import java.awt.dnd.DropTarget
+import java.awt.dnd.DropTargetDragEvent
+import java.awt.dnd.DropTargetDropEvent
+import java.awt.dnd.DropTargetEvent
+import java.awt.dnd.DropTargetListener
+import java.awt.image.BufferedImage
+import java.io.File
+
+/**
+ * Represent data types drag and dropped to an application from outside.
+ */
+@ExperimentalComposeUiApi
+sealed interface DropData {
+    /**
+     * Represents list of files drag and dropped to an application in a raw [java.net.URI] format.
+     */
+    data class FilesList(val rawUris: List<String>) : DropData
+
+    /**
+     * Represents an image drag and dropped to an application.
+     */
+    data class Image(val painter: Painter) : DropData
```

Comment:
This sounds good! But I doubt about mimeType providing, since it can be "image/x-java-image" is it ok?
I am thinking that we can provide a list of mimeTypes like java does

Reply:
>  since it can be "image/x-java-image" is it ok?

Not very pretty and crossplatform (when we port it to others), but If we won't find another way, it can still be useful. People can check individual mimeType's on each platform for different use cases that needed for their aplication, and handle them all.

In the future, when we implement other platforms, we can do research, and change mimeType to something neutral (I think we need to keep this API experimental until we do that)

Reply:
As I checked there is no `application/octet-stream` for every data, at least it won't be clear what the data is for file. Content? File name? etc?, So I would keep mimeType for all, so that user can check it and provide basic interfaces like Image, Text, Uri, FileList